src/GoZipTool/GoZipTool.csproj

```diff
@@ -0,0 +1,54 @@
+<Project Sdk="Microsoft.Build.NoTargets">
+  <PropertyGroup>
+    <TargetFramework>net8.0</TargetFramework>
+    <ResolvedRuntimeIdentifier Condition="'$(RuntimeIdentifier)' != ''">$(RuntimeIdentifier)</ResolvedRuntimeIdentifier>
+  </PropertyGroup>
+
+  <Target Name="SetGoZipPath">
+    <PropertyGroup>
+      <GoOS Condition="'$(ResolvedRuntimeIdentifier)' == 'win-x86'">windows</GoOS>
+      <GoArch Condition="'$(ResolvedRuntimeIdentifier)' == 'win-x86'">386</GoArch>
+
+      <GoOS Condition="'$(ResolvedRuntimeIdentifier)' == 'win-x64'">windows</GoOS>
+      <GoArch Condition="'$(ResolvedRuntimeIdentifier)' == 'win-x64'">amd64</GoArch>
+
+      <GoOS Condition="'$(ResolvedRuntimeIdentifier)' == 'win-arm64'">windows</GoOS>
+      <GoArch Condition="'$(ResolvedRuntimeIdentifier)' == 'win-arm64'">arm64</GoArch>
+
+      <GoOS Condition="'$(ResolvedRuntimeIdentifier)' == 'linux-x64'">linux</GoOS>
+      <GoArch Condition="'$(ResolvedRuntimeIdentifier)' == 'linux-x64'">amd64</GoArch>
+
+      <GoOS Condition="'$(ResolvedRuntimeIdentifier)' == 'osx-x64'">darwin</GoOS>
+      <GoArch Condition="'$(ResolvedRuntimeIdentifier)' == 'osx-x64'">amd64</GoArch>
+
+      <GoOS Condition="'$(ResolvedRuntimeIdentifier)' == 'osx-arm64'">darwin</GoOS>
+      <GoArch Condition="'$(ResolvedRuntimeIdentifier)' == 'osx-arm64'">arm64</GoArch>
+
+      <ExeSuffix Condition="'$(GoOS)' == 'windows'">.exe</ExeSuffix>
+      <GoZipPath>$(IntermediateOutputPath)gozip$(ExeSuffix)</GoZipPath>
+    </PropertyGroup>
+
+    <Error Condition="'$(GoOS)' == '' Or '$(GoArch)' == ''"
+      Text="Unsupported runtime: $(GoOS) - $(GoArch). RID: $(ResolvedRuntimeIdentifier)" />
+
+    <Message Importance="high" Text="GoZip path set to: $(GoZipPath)" />
+  </Target>
+
+  <Target Name="BuildGoZip" Condition="'$(NoBuild)' != 'true'">
+    <PropertyGroup>
+      <GoZipSource>gozip/main.go</GoZipSource>
+    </PropertyGroup>
+
+    <Exec
+      Command="go build -o $(GoZipPath) $(GoZipSource)"
+      EnvironmentVariables="GOOS=$(GoOS);GOARCH=$(GoArch);CGO_ENABLED=0" />
+  </Target>
+
+  <Target Name="CollectGoZipOutput" DependsOnTargets="SetGoZipPath;BuildGoZip" BeforeTargets="AssignTargetPaths" Condition="'$(ResolvedRuntimeIdentifier)' != ''">
+    <Message Importance="high" Text="Collecting GoZip output: $(GoZipPath)" />
```

Comment:
Small polish nit: how does this work with terminal logger? Does it pretty print?

![image](https://github.com/user-attachments/assets/f2b04986-d375-4e8d-bc2f-afa207baad58)

Reply:
Is not pretty, and only shows on `-v:D`

![Screenshot 2025-06-09 at 10 45 55 AM](https://github.com/user-attachments/assets/6b181edc-54b4-4c06-91ad-115846ccfb08)

Which is interesting because with High importance, I shouldn't need to set verbosity to detailed to see it

>Typically, a message set to high is sent when logger verbosity is set to [LoggerVerbosity](https://learn.microsoft.com/en-us/dotnet/api/microsoft.build.framework.loggerverbosity).Minimal or higher. A message set to low is sent when logger verbosity is set to [LoggerVerbosity](https://learn.microsoft.com/en-us/dotnet/api/microsoft.build.framework.loggerverbosity).Detailed.

Reply:
Let me know if any action is needed beside changing importance to low

Reply:
Nothing needed, looks fine to me. I am slightly curious on how to get the GoZipTool.csproj to output a line like:

`GoZipTool succeeded (3.0s) -> /<path>/GoZipTool/debug/gozip`

But that isn't important, or critical to this project. Definitely don't need to spend time figuring that out.

![eye twitch](https://media2.giphy.com/media/v1.Y2lkPTc5MGI3NjExNmh5MzVhOG80Y2FpbGs0cDJwN3ZlbnZ5c2J2ZzNtcDNqeDl5MmJ0NyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/29bKyyjDKX1W8/giphy.gif)